src/source/PeerConnection/SessionDescription.c

```diff
@@ -436,20 +437,47 @@ STATUS populateSingleMediaSection(PKvsPeerConnection pKvsPeerConnection, PKvsRtp
     SPRINTF(pSdpMediaDescription->sdpAttributes[attributeCount].attributeValue, "%d", mediaSectionId);
     attributeCount++;
 
-    switch (pKvsRtpTransceiver->transceiver.direction) {
-        case RTC_RTP_TRANSCEIVER_DIRECTION_SENDRECV:
-            STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "sendrecv");
-            break;
-        case RTC_RTP_TRANSCEIVER_DIRECTION_SENDONLY:
-            STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "sendonly");
-            break;
-        case RTC_RTP_TRANSCEIVER_DIRECTION_RECVONLY:
-            STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "recvonly");
-            break;
-        default:
-            // https://www.w3.org/TR/webrtc/#dom-rtcrtptransceiverdirection
-            DLOGW("Incorrect/no transceiver direction set...this attribute will be set to inactive");
-            STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "inactive");
+
+    if ( pKvsPeerConnection->isOffer ) {
+        switch (pKvsRtpTransceiver->transceiver.direction) {
+            case RTC_RTP_TRANSCEIVER_DIRECTION_SENDRECV:
+                STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "sendrecv");
+                break;
+            case RTC_RTP_TRANSCEIVER_DIRECTION_SENDONLY:
+                STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "sendonly");
+                break;
+            case RTC_RTP_TRANSCEIVER_DIRECTION_RECVONLY:
+                STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "recvonly");
+                break;
+            default:
+                // https://www.w3.org/TR/webrtc/#dom-rtcrtptransceiverdirection
+                DLOGW("Incorrect/no transceiver direction set...this attribute will be set to inactive");
+                STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "inactive");
+        }
+    } else {
+        for ( int i = 0; i < pRemoteSessionDescription->mediaCount; i++ ) {
```

Comment:
Couple issues.
1) Formatting
2) Please declare internal types - in this case UINT32 i
3) Declare the iterator variable up at the top with other variables.